File: refinery-service-java/src/main/java/hu/zoltanterek/worldsim/refinery/model/PatchRequest.java
package hu.zoltanterek.worldsim.refinery.model;

import com.fasterxml.jackson.databind.JsonNode;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

public record PatchRequest(
        @NotBlank String schemaVersion,
        @NotBlank String requestId,
        long seed,
        long tick,
        @NotNull Goal goal,
        @NotNull JsonNode snapshot,
        JsonNode constraints
) {
}

File: WorldSim.RefineryClient/Contracts/PatchRequest.cs
using System.Text.Json.Nodes;
using System.Text.Json.Serialization;

namespace WorldSimRefineryClient.Contracts;

public sealed record PatchRequest(
    [property: JsonPropertyName("schemaVersion")] string SchemaVersion,
    [property: JsonPropertyName("requestId")] string RequestId,
    [property: JsonPropertyName("seed")] long Seed,
    [property: JsonPropertyName("tick")] long Tick,
    [property: JsonPropertyName("goal")] string Goal,
    [property: JsonPropertyName("snapshot")] JsonObject Snapshot,
    [property: JsonPropertyName("constraints")] JsonObject? Constraints
);
